docs: cleanup

joergsteffens opened the pull request:
Cleanup, remove outdated sections.
Remove outdated reference to Mantis Roadmap.
Use the tool black for formatting Python code.

franku commented on docs/manuals/source/Appendix/ReleaseNotes.rst:
```diff
@@ -3,23 +3,17 @@
+.. warning::
 
-The overview about new feature of a release are shown at the :ref:`genindex` of this document.
+      While all the source code is published on `GitHub <https://github.com/bareos/bareos>`_, the releases of packages on http://download.bareos.org is limited to the initial versions of a major release. Later maintenance releases are only published on https://download.bareos.com.
```
```suggestion
      While all the source code is published on `GitHub <https://github.com/bareos/bareos>`_, the releases of packages on http://download.bareos.org is limited to the initial versions of each major release. Later maintenance releases are only published on https://download.bareos.com.
```

franku commented on docs/manuals/source/Appendix/ReleaseNotes.rst:
```diff
@@ -3,23 +3,17 @@
-.. warning::
-
-      While all the source code is published on `GitHub <https://github.com/bareos/bareos>`_, the releases of packages on http://download.bareos.org is limited to the initial versions of a major release. Later maintenance releases are only published on https://download.bareos.com.
+The overview about new feature of a release are shown at the :ref:`genindex` of this document.
```
```suggestion
The overview about new features of a release are shown at the :ref:`genindex` of this document.
```

franku commented on docs/manuals/source/Appendix/ReleaseNotes.rst:
```diff
@@ -3,23 +3,17 @@
 
 This chapter concentrates on things to do when updating an existing Bareos installation.
 
+Please note, that some of the subreleases are only internal development releases.
```
```suggestion
Please note, that some of the subreleases are only non official development releases.
```

joergsteffens replied:
Is "non official" an/our official term for this? In doubt, internal/non official can be removed from the sentense at all.

franku commented on docs/manuals/source/DeveloperGuide/Python.rst:
```diff
@@ -0,0 +1,8 @@
+-------------------------------
+
+All Python code should be formatted using automatic sourcecode formatting.
+For this, the tool `black <https://black.readthedocs.io/en/stable/>`_ have to be used.
```
```suggestion
For this, the tool `black <https://black.readthedocs.io/en/stable/>`_ has to be used.
```

franku commented on docs/manuals/source/DeveloperGuide/api.rst:
```diff
@@ -263,7 +222,7 @@ Example:
     job=BackupClient1pool=Incrementalmessages=Standardclient=client1.example.com-fdstorage=Filewhere=level=Incrementaltype=Backupfileset=SelfTestenabled=1catalog=MyCatalog
 
-This mode is used by BAT.
+This mode has been introduced by BAT and is now deprecated.
```
```suggestion
This mode was introduced by BAT and is now deprecated.
```

pstorz commented on docs/manuals/source/DocumentationStyleGuide/BareosSpecificFormatting/Release.rst:
```diff
@@ -96,6 +96,10 @@ Output:
 
+When refering to a Bareos version,
+always use the version number with three components (e.g. 17.2.7).
+When just refering to the 17.2 release, replace the remaining components by 0, e.g. 17.2.0.
```
Using the 0 as placeholder is ambigous. There could be a version 17.2.0

It would be clear to use 17.2.x or 17.2.*, 17.2.N

franku replied:
I would prefer "17.2" or even "17" That's just as easy as pie.

joergsteffens replied:
That is both not what has been decided in the development meeting from  2019-10-17.

joergsteffens replied:
Removed this lines from the PR, to get it integrated. If required, this can be handled in a separate PR. Until then, everybody will continue to refer to Bareos versions by his own style. Currently used:
```
.. deprecated:: 19
.. deprecated:: 19.2
:sinceVersion:`18.2.0: TLS-PSK`
Bareos Version 18.2
Bareos-18.2
Version 18.2
bareos-14.2.x
bareos-17.2.7
before 18.2
:ref:`bareos-17.2.5`
version 18.2
versionchanged:: 18.2
```

joergsteffens commented:
Rebased on master, some more cleanup and fixes and removed remarks about uniform references to Bareos versions, as there is no agreement about this topic.

joergsteffens commented:
Rebased on current master and resolved conflict, introduced by another PR.

joergsteffens commented:
Rebased to single commit.
